backend/manager/modules/vdsbroker/src/main/java/org/ovirt/engine/core/vdsbroker/VdsManager.java

```diff
@@ -528,6 +528,11 @@ public void handleVdsRecovering() {
     public void updateDynamicData(VdsDynamic dynamicData) {
         vdsDynamicDao.updateIfNeeded(dynamicData);
         cachedVds.setDynamicData(dynamicData);
+        if (cachedVds.getDynamicData().getStatus() == VDSStatus.Maintenance) {
+            cpuTopology = new ArrayList<>();
```

Comment:
this probably should part of `InitVdsOnUpCommand`

Reply:
Since it is just update of VdsManager fields, I would rather put it into `VdsManager.setStatus()`.

Reply:
> This probably should part of `InitVdsOnUpCommand`, where the VM switch from maintenance to UP without clearing it out, just recalling `initAvailableCpus()`.

I would prefer to keep the logic inside the `VdsManager`,  The `cpuTopology` field is internal to `VdsManager`, it makes sense that it is initialized and handled here - there is no need for the outside world to know about it.

The field is cleared to make sure that it is not accidentally overwritten - e.g. when the host is non responsive but still runs the VMs and then it is set to UP state again.

Reply:
> Since it is just update of VdsManager fields, I would rather put it into `VdsManager.setStatus()`.

If we updated only when setting the status, the cachedVds might not have the cpuTopology initialized yet. So I added it also to the refresh() method that updates the cachedVds.